Fix REPORT response from webdav

kobergj opened the pull request:
There were multiple issue with REPORT search response from webdav. Also we want it to be consistent with PROPFIND endpoint.
*   the `remote.php` prefix was missing from the href (added even though not neccessary)
*   the ids were formated wrong, they should look different for shares and spaces.
*   the name of the resource was missing
*   the shareid was missing (for shares)
*   the prop `shareroot` (containing the name of the share root) was missing
*   the  permissions prop was empty

https://github.com/owncloud/web/issues/7557

ownclouders commented:
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-9</strong>](https://drone.owncloud.com/owncloud/ocis/15076/52/1) failed. Further test are cancelled...
<!-- id: pr-4485 -->

kobergj commented:
@ScharfViktor could you help me with the failing tests? We updated the REPORT endpoint to be consistent with the PROPFIND webdav endpoint:
- we added the spaceid to the href (previously it was only storageid!opaqueid, now storageid$spaceid!opaqueid
- we added a `remote.php` prefix to the url
- we removed netencoding from the href (to be consistent with PROPFIND response)

It seems any of these three changes break the tests. 
Do the test expect the href in a certain way? If so can we change that? 
The last two bulletpoints are rather optional but the first one is a must have for proper functionality

ScharfViktor commented:
> @ScharfViktor could you help me with the failing tests? We updated the REPORT endpoint to be consistent with the PROPFIND webdav endpoint:
> 
> * we added the spaceid to the href (previously it was only storageid!opaqueid, now storageid$spaceid!opaqueid
> * we added a `remote.php` prefix to the url
> * we removed netencoding from the href (to be consistent with PROPFIND response)
> 
> It seems any of these three changes break the tests. Do the test expect the href in a certain way? If so can we change that? The last two bulletpoints are rather optional but the first one is a must have for proper functionality

Yes sure. let me switch to your branch and debug it local

ScharfViktor commented:
test failed because: 
1. https://github.com/owncloud/core/blob/master/tests/acceptance/features/bootstrap/WebDav.php#L5319
here we get spaceId. spaceId looks like: `1284d238-aa92-42ce-bdc4-0b0000009157$5ae7fa50-cdf1-43d6-82a5-8f58c65c25f4 `
![Screenshot 2022-08-31 at 16 57 30](https://user-images.githubusercontent.com/84779829/187710766-21f37ab0-9624-4ed5-b519-3a249b342143.png)
but search result give us: `/remote.php/dav/spaces/1284d238-aa92-42ce-bdc4-0b0000009157$5ae7fa50-cdf1-43d6-82a5-8f58c65c25f4!5ae7fa50-cdf1-43d6-82a5-8f58c65c25f4/upload😀 😁/upload😀 😁.txt`

2. here https://github.com/owncloud/core/blob/master/tests/acceptance/features/bootstrap/WebDav.php#L5326
we create $topWebDavPath witch we search in REPORT request. it's ease -> just replace to `"/remote.php/dav/spaces/"`

ScharfViktor commented:
> * the ids were formated wrong, missing the spaceID

PROPFIND request give us: `/remote.php/dav/spaces/1284d238-aa92-42ce-bdc4-0b0000009157$5ae7fa50-cdf1-43d6-82a5-8f58c65c25f4/upload%20folder/`

should be results(REPORT and PROPFIND requests) be same?

ScharfViktor commented:
> should be results(REPORT and PROPFIND requests) be same?

What about this one? It's just that if I change search tests to new implementation-> propfind tests will be failed

kobergj commented:
>No, not exactly. search result returns us 1284d238-aa92-42ce-bdc4-0b0000009157$5ae7fa50-cdf1-43d6-82a5-8f58c65c25f4!5ae7fa50-cdf1-43d6-82a5-8f58c65c25f4 (3 uuid: storageId . "$" . spaceId . "!" spaceId again?? or opaqueid)

Yes exactly. That was a bug on my side. When comparing to `PROPFIND` it should be `storageID$SpaceID/PATH` I'll change it that way.

> Move please failed tests to expected failures. after merging your PR, I create 2 PRs (fix test in core and deleting tests from expected failures in ocis).

Thanks! will do so.

> What about this one? It's just that if I change search tests to new implementation-> propfind tests will be failed
The fields that are in both `PROPFIND` and `REPORT` should be consistent. However `PROPFIND` might contain more fields than `REPORT` does. That means no changes should be done to PROPFIND test, we want `REPORT` to behave like `PROPFIND` not the other way around

kulmann commented on services/search/pkg/search/provider/searchprovider.go:
```diff
@@ -279,3 +307,40 @@ func formatQuery(q string) string {
 }
+
+// NOTE: this converts cs3 to ocs permissions
+// since conversions pkg is reva internal we have no other choice then to duplicate the logic
```
```suggestion
// since conversions pkg is reva internal we have no other choice than to duplicate the logic
```

kulmann commented on changelog/unreleased/fix-search-report.md:
```diff
@@ -0,0 +1,12 @@
+Bugfix: Fix search report
+
+There were multiple issue with REPORT search response from webdav. Also we want it to be consistent with PROPFIND endpoint.
```
```suggestion
There were multiple issues with REPORT search responses from webdav. Also we want it to be consistent with PROPFIND responses.
```

kulmann commented on changelog/unreleased/fix-search-report.md:
```diff
@@ -0,0 +1,12 @@
+
+There were multiple issue with REPORT search response from webdav. Also we want it to be consistent with PROPFIND endpoint.
+*   the `remote.php` prefix was missing from the href (added even though not neccessary)
+*   the ids were formated wrong, they should look different for shares and spaces.
```
```suggestion
*   the ids were formatted wrong, they should look different for shares and spaces.
```

kulmann commented on changelog/unreleased/fix-search-report.md:
```diff
@@ -0,0 +1,12 @@
+*   the name of the resource was missing
+*   the shareid was missing (for shares)
+*   the prop `shareroot` (containing the name of the share root) was missing
+*   the  permissions prop was empty
```
```suggestion
*   the permissions prop was empty
```

kulmann reviewed (approved):
Commented on some typos, but otherwise looks good from what I can tell.

butonic commented on services/webdav/pkg/service/v0/search.go:
```diff
@@ -102,8 +104,39 @@ func multistatusResponse(ctx context.Context, matches []*searchmsg.Match) ([]byt
+	}
 	response := propfind.ResponseXML{
-		Href:     net.EncodePath(path.Join("/dav/spaces/", match.Entity.Ref.ResourceId.StorageId+"!"+match.Entity.Ref.ResourceId.OpaqueId, match.Entity.Ref.Path)),
+		Href:     net.EncodePath(path.Join("/remote.php/dav/spaces/", ref)),
```
why do we have to prefix with the legacy `remote.php`?

kobergj replied:
To be consistent with ocdav PROPFIND endpoint

micbar commented:
@kobergj changelog and pull request id is not matching.

Can we fix that on master?
